libs/designsystem/src/lib/components/icon/icon.component.ts

```diff
@@ -54,13 +57,33 @@ export class IconComponent implements OnChanges {
     }
   }
 
-  constructor(@Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings) {}
+  /**
+   * iconRegistryService: Registry for custom icons.
+   * iconSettings: @deprecated Use KirbyIconRegistryService for adding custom icons.
+   */
+  constructor(
+    private iconRegistryService: KirbyIconRegistryService,
+    @Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings
+  ) {
+    this.combineIconSettings();
+  }
+
+  private combineIconSettings() {
+    const iconSettingsIcons: Icon[] =
+      this.iconSettings && this.iconSettings.icons
+        ? [...this.iconSettings.icons]
+        : [this.defaultIcon];
+
+    this.combinedIconSettings = {
+      icons: [...iconSettingsIcons, ...this.iconRegistryService.getCustomIcons()],
+    } as IconSettings;
```

Comment:
Consider simplyfing:
```suggestion
    const iconSettingsIcons = this.iconSettings ? this.iconSettings.icons : [];
    this.registeredIcons = [...iconSettingsIcons, ...this.iconRegistryService.getCustomIcons()];
```

Reply:
👍